sidekick_core/lib/src/dart_package.dart

```diff
@@ -115,6 +117,27 @@ bool isValidPubPackageName(String name) {
   return _cliNameRegExp.hasMatch(name) && !_keywords.contains(name);
 }
 
+/// Returns the list of all packages in the repository
+List<DartPackage> findAllPackages(Directory directory) {
+  return directory
+      .allSubDirectories((dir) {
+        if (dir.name.startsWith('.')) {
+          // ignore hidden folders
+          return false;
+        }
+        if (dir.name == 'build') {
+          final package = DartPackage.fromDirectory(dir.parent);
+          if (package != null) {
+            // ignore <dartPackage>/build dir
+            return false;
```

Comment:
Unrelated to the PR, but wouldn't this suffice as well?
```dart
if(dir.name == 'build') {
  return false;
}
```
Or what am I missing?

Reply:
No, we want to keep folders named `build` unless they are in the root of a `DartPackage`